### PR TITLE
Add scroll to top button

### PR DESCRIPTION
**Pull request by smyzin**
- added a new hover class for mobile devices
- created new component `ScrollToTopButton`
- implemented a new icon
- updated a footer styles

https://user-images.githubusercontent.com/5114069/193272836-22ac34a2-7778-45ce-a9a0-865d3ac1e4bf.mov

### CODE REVIEW

File: src/icons/ArrowUp.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import classnames, { stroke, strokeWidth } from 'classnames/tailwind'
+
+const pathClassName = classnames(
+  stroke('stroke-primary-light'),
+  strokeWidth('stroke-4')
+)
+
+export default function ({ small }: { small?: boolean }) {
```
Comment (T-Damer) [inline src/icons/ArrowUp.tsx:8]: Let's call this `simpleArrow`, like we did on SC

File: src/icons/ArrowUp.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  if (small)
```
Comment (T-Damer) [inline src/icons/ArrowUp.tsx:9]: You don't need 2 svgs, you need 2 different `div` containers, your `svg` should take whole place in those:
```TS
if (small) return <div className="w-19 h-27><SimpArrow /></div>

return <div className="w-27 h-40"><SimpArrow /></div>
```

File: src/components/ui/ScrollToTopButton.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+const arrowWrapper = (fromMd?: boolean) =>
+  classNamesToString(arrowButton(fromMd), 'hover-button-scroll')
+
+export default function () {
+  const [visible, setVisible] = useState(false)
+  const scrollToTop = () => window.scrollTo({ top: 0, behavior: 'smooth' })
+  const onScroll = () => {
+    setVisible(document.documentElement.scrollTop > 150)
+  }
+  useEffect(() => {
```
Comment (T-Damer) [inline src/components/ui/ScrollToTopButton.tsx:52]: Let's extract this into file, we might need the scroll listener in the future (for example we use it for `NavBar` transparency, maybe we can use only one file for this and Nav?

File: src/components/ui/ScrollToTopButton.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    document.addEventListener('scroll', onScroll)
+    return () => document.removeEventListener('scroll', onScroll)
+  }, [])
+
+  return visible ? (
```
Comment (T-Damer) [inline src/components/ui/ScrollToTopButton.tsx:57]: How about making an animation? Should be simple:
```TS
const scrollContainer = (shown?: boolean) => {
  opacity({'opcaity-0': !shown}),
  pointerEvents({'none': !shown})
}

<Button disabled={!visible}>
```
Something like that, shouldn't be hard

File: src/icons/ArrowUp.tsx (hunk 1 above)
Comment (smyzin) [reply src/icons/ArrowUp.tsx:8]: Done

File: src/index.css [hunk 5]
```diff
@@ -95,6 +95,9 @@ body {
   .hover-secondary:hover {
     color: var(--secondary-light-active-transparent);
   }
+  .hover-button-scroll:hover {
+    filter: drop-shadow(0rem 0rem 4rem var(--primary));
```
Comment (T-Damer) [inline src/index.css]: I think that we can use box-shadow here to save a little speed:
https://stackoverflow.com/questions/27212782/box-shadow-circle

Comment (T-Damer) [reply src/index.css]: But it's up to you, this element shouldn't be that hard to computate

File: src/icons/ArrowUp.tsx (hunk 2 above)
Comment (smyzin) [reply src/icons/ArrowUp.tsx:9]: They are a little bit different with different paths. So it's better to keep both versions

File: src/components/ui/ScrollToTopButton.tsx (hunk 3 above)
Comment (smyzin) [reply src/components/ui/ScrollToTopButton.tsx:56]: Done

File: src/components/ui/ScrollToTopButton.tsx (hunk 4 above)
Comment (smyzin) [reply src/components/ui/ScrollToTopButton.tsx:61]: Did, thx

File: src/index.css (hunk 5 above)
Comment (smyzin) [reply src/index.css]: Not use it anymore

File: src/components/ui/ScrollToTopButton.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+const arrowButton = (shown: boolean, fromMd?: boolean) =>
+  classnames(
+    fromMd ? displayFrom('md') : displayTo('md'),
+    justifyContent('justify-center'),
+    alignItems('items-center'),
+    width('w-12', 'md:w-20'),
+    height('h-12', 'md:h-20'),
+    dropShadow('drop-shadow-primary-dimmed'),
+    borderRadius('rounded-full'),
+    opacity(!shown ? 'opacity-0' : 'opacity-60', 'active:opacity-100'),
```
Comment (T-Damer) [inline src/components/ui/ScrollToTopButton.tsx:46]: Let's do:
```TS
shown ? 'opacity-60' : 'opacity-0'
```

**Review by T-Damer — APPROVED** [review]

**Pull request merged**